backtrace-library/src/main/java/backtraceio/library/models/BacktraceData.java

```diff
@@ -98,103 +97,198 @@ public class BacktraceData {
-        BacktraceLogger.d(LOG_TAG, "Setting annotations");
-        Object exceptionMessage = null;
+    public String getLang() {
+        return lang;
+    }
 
-        if (this.attributes != null &&
-                this.attributes.containsKey("error.message")) {
-            exceptionMessage = this.attributes.get("error.message");
-        }
-        this.annotations = Annotations.getAnnotations(exceptionMessage, complexAttributes);
+    public String getAgent() {
+        return agent;
     }
 
-    /**
-     * Set attributes and add complex attributes to annotations
-     *
-     * @param clientAttributes
-     */
-    private void setAttributes(Map<String, Object> clientAttributes) {
-        BacktraceLogger.d(LOG_TAG, "Setting attributes");
-        BacktraceAttributes backtraceAttributes = new BacktraceAttributes(this.context, this.report,
-                clientAttributes);
-        this.attributes = backtraceAttributes.attributes;
+    public String getSymbolication() {
+        return symbolication;
+    }
 
-        setAnnotations(backtraceAttributes.getComplexAttributes());
+    public long getTimestamp() {
+        return timestamp;
     }
 
-    /**
-     * Set report information such as report identifier (UUID), timestamp, classifier
-     */
-    private void setReportInformation() {
-        BacktraceLogger.d(LOG_TAG, "Setting report information");
-        uuid = report.uuid.toString();
-        timestamp = report.timestamp;
-        classifiers = report.exceptionTypeReport ? new String[]{report.classifier} : null;
-        langVersion = System.getProperty("java.version"); //TODO: Fix problem with read Java version
-        agentVersion = BacktraceClient.version;
+    public String getLangVersion() {
+        return langVersion;
     }
 
-    /**
-     * Set information about all threads
-     */
-    private void setThreadsInformation() {
-        BacktraceLogger.d(LOG_TAG, "Setting threads information");
-        ThreadData threadData = new ThreadData(report.diagnosticStack);
-        this.mainThread = threadData.getMainThread();
-        this.threadInformationMap = threadData.threadInformation;
-        SourceCodeData sourceCodeData = new SourceCodeData(report.diagnosticStack);
-        this.sourceCode = sourceCodeData.data.isEmpty() ? null : sourceCodeData.data;
+    public String getAgentVersion() {
+        return agentVersion;
+    }
+
+    public Map<String, String> getAttributes() {
+        return attributes;
+    }
+
+    public String getMainThread() {
+        return mainThread;
+    }
+
+    public String[] getClassifiers() {
+        return classifiers;
+    }
+
+    public Map<String, Object> getAnnotations() {
+        return annotations;
+    }
+
+    public Map<String, SourceCode> getSourceCode() {
+        return sourceCode;
+    }
+
+    public BacktraceReport getReport() {
+        return report;
+    }
+
+
+    public static class Builder {
```

Comment:
This class is already huge. We should move builder to another file.

I think we can also drop context reference in the BacktraceAttributes - some information can be just generated on the app startup, or reference to windowsmanager can be passed from the context in the backtraceattribtues class creation. There is a way to drop the reference to the context here.

Reply:
it's builder pattern and it should be combined with BacktraceData 

https://blogs.oracle.com/javamagazine/post/exploring-joshua-blochs-builder-design-pattern-in-java
https://www.digitalocean.com/community/tutorials/builder-design-pattern-in-java

Reply:
We can not generate all of the necessary informations on startup that is why we can't easily drop context from here

Things which we can get on startup:
- application name
- application package 
- application version

Things which we can't get on startup:
- screen brightness and orientation
- memory
- gps/nfc/bluetooth/airplane mode/storage etc.

Reply:
we will work on that and revalidate whole BacktraceAttributes mechanism as part of BT-3218